test/torch/tensors/test_large_precision.py

```diff
@@ -233,8 +233,21 @@ def test_mod():
         (torch.tensor([2000.0123458910]), torch.tensor([2000.0123458910])),
     ],
 )
-def test_types(x, expected):
-    enlarged = x.fix_prec(internal_type=torch.int16, precision_fractional=128)
-    restored = enlarged.float_precision()
-    # And now x and restored must be the same
-    assert torch.all(torch.eq(expected, restored))
+class TestParametrized:
+    def test_types(self, x, expected):
+        enlarged = x.fix_prec(internal_type=torch.int16, precision_fractional=128)
+        restored = enlarged.float_precision()
+        # And now x and restored must be the same
+        assert torch.all(torch.eq(expected, restored))
+
+    def test_share_and_get(self, workers, x, expected):
+        alice, bob, james = (workers["alice"], workers["bob"], workers["james"])
+
+        enlarged = x.fix_prec(internal_type=torch.int16, precision_fractional=128)
+        expected = expected.fix_prec(internal_type=torch.int16, precision_fractional=128)
+
+        shared = enlarged.share(alice, bob, crypto_provider=james)
+
+        result = shared.get()
```

Comment:
Are operations supported on LPT shared data?
Is so, add tests for them to make sure everything goes fine :)

Reply:
This one is tricky. Working on it

Reply:
A first for `add` is now ready. Let's go through it, polish it and once we're happy I can apply the same to the other operations.